server/src/utils.ts

```diff
@@ -249,16 +249,34 @@ function generateDiagnostic(severity: DiagnosticSeverity,
   return diagnostic
 }
 
-function calculateElementPositions(element: any): Position[] {
+export function calculateElementPositions(element: any): Position[] {
   // Calculate positions accounting for the zero-based convention used by
   // vscode
   const startPosition: Position = {
     line: element.lineNumber - 1,
     character: element.columnNumber - 1
   }
-  const endPosition: Position = {
-    line: element.nextSibling.lineNumber - 1,
-    character: element.nextSibling.columnNumber - 1
+  const elementSibling = element.nextSibling
+  let endPosition: Position
+
+  // Use the sibling element to get accurate position of the end of the
+  // input element if it's available. Otherwise we'll use the position and
+  // content of the last attribute to establish an end position
+  if (elementSibling === null) {
+    const elementAttributes = element.attributes
+    const finalAttribute = elementAttributes[elementAttributes.length - 1]
```

Comment:
Oh wait, what if the element has no next sibling and no attributes?

Reply:
Ah, good catch. The current callers of this function don't have that happen because of the xpath queries they use, but in general it could. I'll add a test case + fallback to just the tag.

Reply:
@tomjw64 - [Fixed](https://github.com/openstax/poet/pull/22/commits/d2390a9d195b5e291d515b76de57b74102756743)